src/components/augmentedReality/AugmentedRealityView.js

```diff
@@ -76,41 +71,78 @@ export const AugmentedRealityView = ({ sceneNavigator }) => {
   );
 };
 
+// eslint-disable-next-line complexity
 const ViroSoundAnd3DObject = (item) => {
   const {
     isObjectLoading,
     isStartAnimationAndSound,
     object,
-    position,
-    rotation,
-    scale,
     setIsObjectLoading,
     setIsStartAnimationAndSound
   } = item;
 
+  // if the `chromaKeyFilteredVideo` prop is undefined, the default colour of GreenScreen is defined
+  ViroMaterials.createMaterials({
+    chromaKeyFilteredVideo: {
+      chromaKeyFilteringColor: object?.mp4?.chromaKeyFilteredVideo || '#00FF00'
+    }
+  });
+
   return (
     <>
-      {!!object.mp3 && !isObjectLoading && (
-        <ViroSound
-          source={{ uri: object.mp3 }}
-          paused={!isStartAnimationAndSound}
-          onFinish={() => setIsStartAnimationAndSound(false)}
+      {!!object?.mp3 &&
+        !isObjectLoading &&
+        (object?.mp3?.isSpatialSound ? (
+          <ViroSpatialSound
+            source={{ uri: object?.mp3?.uri }}
+            paused={!isStartAnimationAndSound}
+            onFinish={() => setIsStartAnimationAndSound(false)}
+            maxDistance={object?.mp3?.maxDistance}
+            minDistance={object?.mp3?.minDistance}
+            position={object?.mp3?.position}
+            rolloffModel={object?.mp3?.rolloffModel}
+          />
+        ) : (
+          <ViroSound
+            source={{ uri: object?.mp3?.uri }}
+            paused={!isStartAnimationAndSound}
+            onFinish={() => setIsStartAnimationAndSound(false)}
+          />
+        ))}
+
+      {!!object?.mp4 && !isObjectLoading && (
+        <ViroVideo
+          loop
+          materials={['chromaKeyFilteredVideo']}
+          position={object?.mp4?.position}
+          rotation={object?.mp4?.rotation}
+          scale={object?.mp4?.scale}
+          source={{ uri: object?.mp4?.uri }}
+        />
+      )}
+
+      {!!object?.image && !isObjectLoading && (
```

Comment:
you are checking `!isObjectLoading` every time. i would suggest to check that once before all the several objects.

Reply:
Yes, you're right. 
https://github.com/ikuseiGmbH/smart-village-app-app/pull/634/commits/c69f385584050e19f46a35da65fddf329cc59a5c